### PR TITLE
Fixed Cyborg Self-Repair giving action buttons incorrectly

### LINKED ISSUE #39037: Borg self-repair module can be toggled by the last person who touched it if ejected by module reset.
Issue reported from  Round ID: 90820 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

Reproduction steps: Give borg self-repair module. Module reset the borg to eject the module. Re-insert the module. You can now toggle it, as well as the person who inserted it.

Notes: I am not sure if the repair module has to be activated once or not before this will work. It may be if there's some sort of initialization on first run.

### PR DESCRIPTION
**Pull request by zxaber**
Borg self repair repair

:cl: Zxaber
fix: Cyborg Self-Repair item no longer attaches an action button to anyone that picks it up under certain circumstances. You can no longer remote-control a borg's Self-Repair.
/:cl:

Fixes #39037

Fun fact, you can use this bug to remotely tell if the borg is alive, as the self repair module will not stay on if the borg is offline.

The Self-Repair item pops out with a different sprite (the mech's self-repair drone) than the standard circuit board, so it seems there is a "mode" it can be in to actually set the button. Not sure, still learning to read code. Anyway, the item's code has calls for when it's dropped, and you can actually throw it on the ground to remove the bugged button, which reverts it back to being a circuit board. My guess is upgrades removed from a borg were originally dropped at some point, before the code changed to having them simply moved out. This fix simply checks each removed upgrade to see if it has a "dropped" proc defined, and runs it if so. Currently, only the Self-Repair does, but if other upgrades in the future copy the code, they'll get the same treatment.

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -990,6 +990,8 @@
 	// Remove upgrades.
 	for(var/obj/item/I in upgrades)
 		I.forceMove(get_turf(src))
+		if (hascall(I,"dropped"))
```
Comment (AnturK) [inline code/modules/mob/living/silicon/robot/robot.dm]: Instead of this hack, just call `deactivate()` on the upgrades

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (zxaber) [reply code/modules/mob/living/silicon/robot/robot.dm]: Thing is, the selfrepair/dropped() proc (in robot_upgrades.dm) has several other things it does first before calling deactivate_sr() (dropped leads to check_dropped, which eventually leads to deactivate_sr). Should I just move all that over to deactivate_sr()? 

Additionally (and I don't mean this as an attack or insult) how is calling deactivate_sr() less hacky than calling dropped()? Either way, it's an extra call that currently only fixes a single edge case, right?

Comment (AnturK) [reply code/modules/mob/living/silicon/robot/robot.dm]: These are just wrappers, there's no change needed in the repair itself.

Comment (zxaber) [reply code/modules/mob/living/silicon/robot/robot.dm]: I still don't think I'm super clear on what you were suggesting. Neither `deactivate()` nor `deactivate_sr()` compile when swapped for `dropped()`, so I might be misunderstanding.

Removed the unneeded check, though

Comment (AnturK) [reply code/modules/mob/living/silicon/robot/robot.dm]: You need change the iterated path to `/obj/item/borg/upgrade`, and switch the call to `deactivate(src)`, sorry for late response.

**Pull request merged**